pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/periods/PeriodParser.scala

```diff
@@ -0,0 +1,199 @@
+    (qualified(year) to calendarDate).toInstantRange |
+      (qualified(year) to monthAndYear).toInstantRange |
+      (year to calendarDate).toInstantRange |
+      (year to monthAndYear).toInstantRange |
+      (year to century).toInstantRange |
+      (qualified(year) to century).toInstantRange |
+      (year to qualified(century)).toInstantRange |
+      (qualified(year) to qualified(century)).toInstantRange |
+      (qualified(year) to qualified(decade)).toInstantRange |
+      (qualified(year) to decade).toInstantRange |
+      (year to qualified(decade)).toInstantRange |
+      (year to decade).toInstantRange |
+      (year to year).toInstantRange |
+      (qualified(year) to year).toInstantRange |
+      (year to qualified(year)).toInstantRange |
+      (qualified(year) to qualified(year)).toInstantRange
+
+  def decadeToDate[_: P]: P[InstantRange] =
+    (qualified(decade) to qualified(decade)).toInstantRange |
+      (qualified(decade) to decade).toInstantRange |
+      (decade to qualified(decade)).toInstantRange |
+      (decade to decade).toInstantRange |
+      (qualified(decade) to qualified(year)).toInstantRange |
+      (qualified(decade) to year).toInstantRange |
+      (decade to qualified(year)).toInstantRange |
+      (decade to year).toInstantRange
+
+  def noopQualifier[_: P]: P[Unit] = Lex.qualifier map (_ => ())
+
+  // We can infer a century intention for some numbers, eg for
+  // 14 in the string "14th-15th century"
+  def inferredCentury[_: P]: P[Century] =
+    P(Lex.int.filter(_ <= 999) ~ Lex.ordinalSuffix.? map { n =>
+      Century(n - 1)
+    })
+
+  def century[_: P]: P[Century] = Lex.century map Century
+
+  def decade[_: P]: P[CenturyAndDecade] = Lex.decade map { year =>
+    CenturyAndDecade(century = year / 100, decade = (year % 100) / 10)
+  }
+
+  // A year range is a string like 1994-5 or 1066-90
+  def yearRange[_: P]: P[FuzzyDateRange[Year, Year]] =
+    P(year ~ ws.? ~ "-" ~ ws.? ~ digitRep(1, 2) map {
+      case (year @ Year(y), n) if n < 10 =>
+        FuzzyDateRange(year, Year(y - (y % 10) + n))
+      case (year @ Year(y), n) if n < 100 =>
+        FuzzyDateRange(year, Year(y - (y % 100) + n))
+    })
+
+  def monthAndDay[_: P]: P[MonthAndDay] =
+    dayFollowedByMonth | monthFollowedByDay
+
+  def dayFollowedByMonth[_: P]: P[MonthAndDay] =
+    P(writtenDay ~ ws ~ writtenMonth map { case (d, m) => MonthAndDay(m, d) })
+
+  def monthFollowedByDay[_: P]: P[MonthAndDay] =
+    P(writtenMonth ~ ws ~ writtenDay map MonthAndDay.tupled)
+
+  def yearDivision[_: P]: P[FuzzyDateRange[MonthAndYear, MonthAndYear]] =
+    seasonYear | lawTermYear
+
+  def seasonYear[_: P]: P[FuzzyDateRange[MonthAndYear, MonthAndYear]] =
+    P((Lex.season map {
+      case "spring" => (3, 5)
+      case "summer" => (6, 8)
+      case "autumn" => (9, 11)
+      // Winter YEAR refers to the year in which the winter starts
+      // https://www.metoffice.gov.uk/weather/learn-about/weather/seasons/winter/when-does-winter-start
```

Comment:
nice comment

Reply:
Realised that while I thought this was the case had absolutely no evidence for it and briefly convinced myself that since there are more winter months in the following year it was the other way round.